File: frontend/src/types/document.ts
export enum DocumentType {
  CV = 'cv',
  JOB_OFFER = 'job_offer'
}

export interface Document {
  id: number;
  user_id: number;
  document_type: DocumentType;
  original_filename: string;
  file_size: number;
  mime_type: string;
  uploaded_at: string;
  extracted_text?: string;
}

export interface DocumentUploadResponse {
  id: number;
  original_filename: string;
  document_type: DocumentType;
  file_size: number;
  uploaded_at: string;
  message: string;
}

File: backend/app/schemas/document.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict
from enum import Enum

class DocumentType(str, Enum):
    CV = "cv"
    JOB_OFFER = "job_offer"

class DocumentBase(BaseModel):
    document_type: DocumentType
    original_filename: str
    file_size: int
    mime_type: str

class DocumentCreate(DocumentBase):
    user_id: int
    stored_filename: str
    file_path: str
    extracted_text: Optional[str] = None

class DocumentResponse(DocumentBase):
    id: int
    user_id: int
    uploaded_at: datetime
    extracted_text: Optional[str] = None
    
    model_config = ConfigDict(from_attributes=True)

class DocumentUploadResponse(BaseModel):
    id: int
    original_filename: str
    document_type: DocumentType
    file_size: int
    uploaded_at: datetime
    message: str